studio/LonaStudio/VersionControl/Git.swift

```diff
@@ -79,81 +101,97 @@ public enum Git {
 // MARK: Sync
 
 extension Git.Client {
-    public func initRepo() -> Result<String, NSError> {
+    public func initRepo() -> Result<String, ProcessError> {
         return run(arguments: ["init"])
     }
 
-    public func addAllFiles() -> Result<String, NSError> {
+    public func addAllFiles() -> Result<String, ProcessError> {
         return run(arguments: ["add", "."])
     }
 
-    public func commit(message: String) -> Result<String, NSError> {
+    public func commit(message: String) -> Result<String, ProcessError> {
         return run(arguments: ["commit", "-m", message])
     }
 
-    public func addRemote(name: String, url: URL) -> Result<String, NSError> {
+    public func addRemote(name: String, url: URL) -> Result<String, ProcessError> {
         return run(arguments: ["remote", "add", name, url.absoluteString])
     }
 
-    public func push(repository: String, refspec: String) -> Result<String, NSError> {
-        return run(arguments: ["push", repository, refspec])
+    public func push(repository: String, refspec: String) -> Result<String, GitError> {
+        return run(arguments: ["push", repository, refspec]).mapError({ error in
+            if error.code == 128,
+                let errorString = error.errorOutput.utf8String(),
+                (errorString.contains("git@github.com: Permission denied (publickey)") || errorString.contains("Host key verification failed")) {
```

Comment:
in dugite (what GitHub is using), they just check that the code is !== 0 and that's their regex: https://github.com/desktop/dugite/blob/e447c5ef95b2cfd13e98a4d85f89e7906f82dbdc/lib/errors.ts#L60-L63

Reply:
:+1: copying theirs